main.go

```diff
@@ -93,6 +93,13 @@ func manageIncident(event *corev2.Event) error {
 	if len(summary) > 1024 {
 		summary = summary[:1024]
 	}
+	log.Printf("Incident Summary: %s", summary)
+
+	// "The maximum permitted length of PG event is 512 KB. Let's limit check output to 256KB to prevent triggering a failed send"
+	if len(event.Check.Output) > 256000 {
+		log.Printf("Warning Incident Payload Truncated!")
+		event.Check.Output = "WARNING Truncated:i\n" + event.Check.Output[:256000] + "..."
+	}
```

Comment:
I don't think that's quite going to do it. The hefty portion of the message could be the entity, which could contain plenty of labels and annotations, process information, network adapters, etc, etc.

I think the best thing to do is serialize the event, send it to PagerDuty no matter what the size, and check for a 413 return code. If that happens, construct a minimal event and entity, simply to trigger the alert.
```
// say there is an event called largeEvent, and we sent it off already
if resp.StatusCode == 413 { 
  event := &corev2.Event{
    ObjectMeta: corev2.ObjectMeta{
      Namespace: largeEvent.Namespace,
      Name: largeEvent.Name,
    },
    Check: &corev2.Check{
      ObjectMeta: corev2.ObjectMeta{
        Namespace: largeEvent.Check.Namespace,
        Name: largeEvent.Check.Name,
      },
      Status: 2,
      Output: "PD rejected event, larger than 512 KB",
    },
    Entity: &corev2.Entity{
      ObjectMeta: corev2.ObjectMeta{
        Namespace: largeEvent.Entity.Namespace,
        Name: largeEvent.Entity.Name,
      },
    },
  }
  b, err := json.Marshal(event)
  ...
  // send event to pagerduty
}
```

We could check the size of each label and annotation, and add those too, but I think that's enough information to diagnose and repair a now-faulty check.